addons/sourcemod/scripting/scp_sf.sp

```diff
@@ -1981,19 +1984,60 @@ public Action TF2Items_OnGiveNamedItem(int client, char[] classname, int index,
 			return Plugin_Continue;
 	}
 
-	switch(index)
+	if (Client[client].Class == Classes_GetByName("dboi") && StrContains(classname, "tf_wearable") != -1 && CvarAllowCosmetics.BoolValue)
 	{
-		case 493, 233, 234, 241, 280, 281, 282, 283, 284, 286, 288, 362, 364, 365, 536, 542, 577, 599, 673, 729, 791, 839, 5607:  //Action slot items
+		int newItemRegionMask = TF2Econ_GetItemEquipRegionMask(index);
+		int newItemRegionGroupBits = TF2Econ_GetItemEquipRegionGroupBits(index);
+
+		StringMap regions = TF2Econ_GetEquipRegionGroups();
+		StringMapSnapshot snapshot = regions.Snapshot();
+
+		// Multiple groups can share the same group bit, so test each group name
+		for (int i = 0; i < snapshot.Length; i++)
 		{
-			return Plugin_Continue;
+			char buffer[16];
+			snapshot.GetKey(i, buffer, sizeof(buffer));
+
+			int bit;
+			if (regions.GetValue(buffer, bit) && (newItemRegionGroupBits >> bit) & 1)
+			{
+				// Disallow shirts and pants to avoid obscuring the D-Class colors
+				if (StrEqual(buffer, "shirt") || StrEqual(buffer, "pants"))
+					return Plugin_Handled;
+			}
 		}
-		case 125, 134, 136, 138, 260, 470, 640, 711, 712, 713, 1158:  //Special hats
+		delete snapshot;
+		delete regions;
+
+		// Remove any wearable that has a conflicting equip_region
+		for (int wbl = 0; wbl < TF2Util_GetPlayerWearableCount(client); wbl++)
 		{
-			ClassEnum class;
-			if(Classes_GetByIndex(Client[client].Class, class) && class.Human)
-				return Plugin_Continue;
+			int wearable = TF2Util_GetPlayerWearable(client, wbl);
+			if (wearable == -1)
```

Comment:
Would've used `INVALID_ENT_REFERENCE` instead of `-1` on invalid entity

Reply:
Documentation for this function says `-1` so I'll keep it